### PR TITLE
Docs: Add transformations documentation

### PR DESCRIPTION
**Pull request by oddlittlebird**
This topic updates the side menu, adds the Transformations topic, and adds the Calculations list reference topic.

**Special notes for your reviewer**:
Images are in PR https://github.com/grafana/website/pull/1466

@dprokop , I think the Join transform result image is bugged. 

@torkelo and @daniellee , I need definitions for the calculations in calculation-list.md. 

All, feel free to expand on examples, suggestions, and use cases.

### CODE REVIEW

**Comment by torkelo** [comment]
Yea the join image should be updated to use time series and join by time (it's what it is mainly designed/optimized for right now). Posted an image in the images PR

File: docs/sources/panels/transformations.md [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+type = "docs"
+[menu.docs]
+identifier = "transformations"
+parent = "panels"
+weight = 300
++++
+
+# Transformations
+
+> **Note:** This documentation refers to a Grafana 7.0 beta feature. This documentation will be frequently updated to reflect updates to the feature, and it will probably be broken into smaller sections when the feature moves out of beta.
```
Comment (dprokop) [inline docs/sources/panels/transformations.md:12]: +1

File: docs/sources/panels/transformations.md [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+
+This page explains what transformations in Grafana are and how to use them.
+
+Transformations process the result set before it’s passed to the visualization. You access transformations in the Transform tab of the Grafana panel editor.
```
Comment (dprokop) [inline docs/sources/panels/transformations.md:16]: Some times we refer to panel editor as `panel editor` and sometimes as `Panel editor`. Think we should align on this one.

File: docs/sources/panels/transformations.md [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+
+Transformations allow you to rename fields, join separate time series together, do math across queries, and more. For users\, with large dashboards or with heavy queries, being able to reuse the query result from one panel in another panel can be a huge performance gain.
```
Comment (dprokop) [inline docs/sources/panels/transformations.md:18]:
```suggestion
Transformations allow you to rename fields, join separate time series together, do math across queries, and more. For users, with large dashboards or with heavy queries, being able to reuse the query result from one panel in another panel can be a huge performance gain.
```

File: docs/sources/panels/transformations.md [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+  - Stat
+  - Gauge
+  - Bar gauge
+  - Table
+  - Heatmap
+  - Logs
+
+## Apply a transformation
+
+Transformations are available from the Transform tab in the bottom pane of the Panel Editor, next to the Queries tab.
```
Comment (dprokop) [inline docs/sources/panels/transformations.md:44]: Again, panel editor, Panel editor or Panel Editor? :)

File: docs/sources/panels/transformations.md [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+1. Click the **Transform** tab.
+1. Click a transformation to select it. 
+  
+   A transformation row appears that allows you to configure the transformation options.
+
+   Click **Add transformation** to apply another transformation. Keep in mind that the next transformation acts on the result set returned by the previous transformation.
+
+   If you have trouble, click the bug icon to [debug your transformations](#debug-transformations).
+
+   Click the trash can icon to permanently remove a transformation.
```
Comment (dprokop) [inline docs/sources/panels/transformations.md:56]:
```suggestion
   Click the trash can icon to remove a transformation.
```
It's permanent only if you save the dashboard.

File: docs/sources/panels/transformations.md [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+- Add field from calculation - Create new fields that are the result of result set row calculation.
+- Labels to fields - Group a series by time and return labels as fields.
+
+Keep reading for detailed descriptions of each type of transformation and the options available for each, as well as suggestions for how to use them.
+
+### Reduce
+
+Apply a _Reduce_ transformation when you want to simplify your results down to one value. Reduce basically removes time component. If visualized as a table, it reduces a column down to one row (value).
+
+In the **Calculations** field, enter one or more calculation types. Click to see a list of calculation choices. For information about available calculations, refer to the [Calculation list]({{< relref "calculations-list.md" >}}).
```
Comment (dprokop) [inline docs/sources/panels/transformations.md:78]:
```suggestion
In the **Calculations** field, select one or more calculation types. Click to see a list of calculation choices. For information about available calculations, refer to the [Calculation list]({{< relref "calculations-list.md" >}}).
```

File: docs/sources/panels/transformations.md [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+I applied a transformation to join the query results using the time field. Now I can run calculations, combine, and organize the results in this new table.
+
+{{< docs-imagebox img="/img/docs/transformations/join-fields-after-7-0.png" class="docs-image--no-shadow" max-width= "1100px" >}}
+
+### Add field from calculation
+
+Use this transformation to add a new field calculated from two other fields. Each transformation allows you to add one new field.
+
+- **Mode -** Select a mode:
+  - **Reduce row -** NEED DEFINITION
```
Comment (dprokop) [inline docs/sources/panels/transformations.md:160]:
```suggestion
  - **Reduce row -** Apply selected calculation on each row of selected fields independently.
```

File: docs/sources/panels/transformations.md [hunk 8]
```diff
@@ -0,0 +1,193 @@
...
+  - **Binary option -** NEED DEFINITION
```
Comment (dprokop) [inline docs/sources/panels/transformations.md:161]:
```suggestion
  - **Binary option -** Apply basic math operation(sum, multiply, etc) on values in a single row from two selected fields.
```

**Review by dprokop — APPROVED** [review]
I think this is great start of the transformations docs!

File: docs/sources/panels/calculations-list.md [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+| Difference |    |
+| Distinct count |    |
+| First (not null) |    |
+| Max |    |
+| Mean |    |
+| Min |    |
+| Min (above zero) |    |
+| Range |    |
+| Step |    |
+| Total |    |
```
Comment (dprokop) [inline docs/sources/panels/calculations-list.md:32]:
```suggestion
| All nulls | true when all values are null|
| All zeros | true when all values are 0 |
| Change count | number of times the field's value changes |
| Count | number of values in a field|
| Delta | cumulative change in value |
| Difference | difference between first and last value of a field|
| Distinct count | number of unique values in a field |
| First (not null) | first, not null value in a field|
| Max | maximum value of a field |
| Mean | mean value of all values in a field |
| Min | minimum value of a field|
| Min (above zero) | minimum, positive value of a field|
| Range | difference between maximum and minimum values of a field|
| Step | minimal interval between values of a field |
| Total | sum of all values in a field |
```

**Comment by dprokop** [comment]
> @torkelo and @daniellee , I need definitions for the calculations in calculation-list.md.

@oddlittlebird done:)

**Comment by oddlittlebird** [comment]
> Yea the join image should be updated to use time series and join by time (it's what it is mainly designed/optimized for right now). Posted an image in the images PR

@torkelo , please make sure my updated text in the Join transformation is correct.

File: docs/sources/panels/transformations.md (hunk 2 above)
Comment (oddlittlebird) [reply docs/sources/panels/transformations.md:16]: It is the Edit Panel window, so lowercase "panel editor." Will reassess if the name changes in the UI. I thought it was going to change to "Panel edit" in the UI, which was close enough to "Panel editor" for me to justify the capitalization.

File: docs/sources/panels/transformations.md (hunk 6 above)
Comment (oddlittlebird) [reply docs/sources/panels/transformations.md:78]: "enter" is correct, because you can type them or select.

**Comment by jaakoob** [comment]
The pictures showing the before and after merge state of the time series are swapped (`/img/docs/transformations/time-series-before-merge-7-1.png` and `/img/docs/transformations/time-series-after-merge-7-1.png`). The naming is correct in this repository, so i would assume, that the image names were swapped accidentally in grafana/website. Since grafana/website is not public i kindly ask you to fix the issue there.

**Comment by oddlittlebird** [comment]
@dprokop, can you help improve the images?

**Pull request merged**